test/e2e/framework/pod/client/client.go

```diff
@@ -0,0 +1,286 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"sync"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+
+	v1core "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/kubectl/pkg/scheme"
+	"k8s.io/kubectl/pkg/util/podutils"
+)
+
+const (
+	// DefaultPodDeletionTimeout is the default timeout for deleting pod
+	DefaultPodDeletionTimeout = 3 * time.Minute
+
+	// the status of container event, copied from k8s.io/kubernetes/pkg/kubelet/events
+	killingContainer = "Killing"
+
+	// the status of container event, copied from k8s.io/kubernetes/pkg/kubelet/events
+	failedToCreateContainer = "Failed"
+
+	// the status of container event, copied from k8s.io/kubernetes/pkg/kubelet/events
+	startedContainer = "Started"
+
+	// it is copied from k8s.io/kubernetes/pkg/kubelet/sysctl
+	forbiddenReason = "SysctlForbidden"
+)
+
+// ImagePrePullList is the images used in the current test suite. It should be initialized in test suite and
+// the images in the list should be pre-pulled in the test suite.  Currently, this is only used by
+// node e2e test.
+var ImagePrePullList sets.String
+
+// PodClient is a struct for pod client.
+type PodClient struct {
+	f FrameworkShim
+	v1core.PodInterface
+}
+
+// FrameworkShim exposes the same methods as test/e2e/framework, this is
+// done for compatibility as work on #81245 continues.
+type FrameworkShim interface {
+	GetClientSet() clientset.Interface
+	GetNamespace() *v1.Namespace
+}
+
+// MakePodClient is a convenience method for getting a pod client interface in the framework's namespace,
+// possibly applying test-suite specific transformations to the pod spec, e.g. for
+// node e2e pod scheduling.
+func MakePodClient(f FrameworkShim) *PodClient {
+	return &PodClient{
+		f:            f,
+		PodInterface: f.GetClientSet().CoreV1().Pods(f.GetNamespace().Name),
```

Comment:
just curious, why isn't this implemented reusing `MakePodClientNS`, like
```
func MakePodClient(f FrameworkShim) *PodClient {
    return MakePodClientNS(f, f.GetNamespace().Name)
}
```
?

Reply:
it was like that in f/pods.go but I can change it

Reply:
I'm fine with any approach, I don't have strong opinions. Just curious.